From fe85ad4bd200a4417e5f6af10d01f0800b6c0dd2 Mon Sep 17 00:00:00 2001
From: Allen Wild <allenwild93@gmail.com>
Date: Sun, 6 May 2018 18:50:07 -0400
Subject: [PATCH] xz: use update-alternatives

Installing xz and busybox together may cause conflicts for xz, xzcat,
unxz, and their lzma variants. In the default configuration, xzcat is
silently replaced with a symlink to busybox. If busybox is compiled with
CONFIG_XZ=y, its postinst fails during do_rootfs.

Using update-alternatives to xz handles these conflicts properly.

Signed-off-by: Allen Wild <allenwild93@gmail.com>
---
 meta/recipes-extended/xz/xz_5.2.3.bb | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/meta/recipes-extended/xz/xz_5.2.3.bb b/meta/recipes-extended/xz/xz_5.2.3.bb
index c0f8b82f64..65168920d1 100644
--- a/meta/recipes-extended/xz/xz_5.2.3.bb
+++ b/meta/recipes-extended/xz/xz_5.2.3.bb
@@ -31,6 +31,11 @@ PACKAGES =+ "liblzma"
 
 FILES_liblzma = "${libdir}/liblzma*${SOLIBS}"
 
+inherit update-alternatives
+ALTERNATIVE_PRIORITY = "100"
+ALTERNATIVE_${PN} = "xz xzcat unxz \
+                     lzma lzcat unlzma"
+
 BBCLASSEXTEND = "native nativesdk"
 
 export CONFIG_SHELL="/bin/sh"
-- 
2.17.0

